cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -1514,13 +1519,17 @@ void reader::impl::allocate_columns(read_mode mode, size_t skip_rows, size_t num
         // we're going to start null mask as all valid and then turn bits off if necessary
         out_buf.create_with_mask(
           out_buf.type.id() == type_id::LIST && l_idx < max_depth ? num_rows + 1 : num_rows,
-          cudf::mask_state::ALL_VALID,
+          cudf::mask_state::UNINITIALIZED,
           _stream,
           _mr);
+        memset_bufs.push_back(cudf::device_span<uint64_t>(static_cast<uint64_t*>(out_buf.data()),
+                                                          out_buf.data_size() / sizeof(uint64_t)));
+        nullmask_bufs.push_back(
+          cudf::device_span<uint64_t>(reinterpret_cast<uint64_t*>(out_buf.null_mask()),
+                                      out_buf.null_mask_size() / sizeof(uint64_t)));
```

Comment:
I don't think the math here is right.  If `out_buf.data_size()` is 1, you're going to get a span of size 0.  This should always be rounded up.  Also, a comment on why it's safe to do this (the allocation padding as above).